### PR TITLE
`networkx`, `scipy`, `ipywidgets`: Update version ranges in `conda.txt`

### PR DESCRIPTION
**Pull request by mkoeppe**
Just one commit split out from #35593.

### :memo: Checklist

- [x] The title is concise, informative, and self-explanatory.
- [ ] The description explains in detail what this PR is about.
- [x] I have linked a relevant issue or discussion.
- [ ] I have created tests covering the changes.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

**Review by tobiasdiez — APPROVED** [review]
LGTM, thanks!

Could you have a look a the failing tests in the macos workflow? They are not related to this PR, but randomly occur.

**Comment by mkoeppe** [comment]
I'm taking this as a "positive review".

File: build/pkgs/networkx/distros/conda.txt [hunk 1]
```diff
@@ -1 +1 @@
-networkx<3.0,>=2.4
+networkx<3.2,>=2.4
```
Comment (tornaria) [inline build/pkgs/networkx/distros/conda.txt:1]:
```suggestion
networkx>=2.4
```
I suggest this change which depends on #36544.

Comment (mkoeppe) [reply build/pkgs/networkx/distros/conda.txt:1]: This PR is already on Volker's branch. I'd suggest to merge it into #36544 and make the change there

**Pull request merged**